### PR TITLE
Add 'WriteConnectionSecretToRef' to API spec

### PR DESCRIPTION
**Pull request by ccremer**
## Summary

* Adds `spec.writeConnectionSecretToRef` to the API. This field accepts a string name so that later a `Secret` in the instance namespace can be created with the connection details. This mechanism is similar to how Crossplane handles it.
* Instead of making this field `Required` I opted for a default that sets `spec.writeConnectionSecretToRef` to the same name as `metadata.name` if not provided.

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] PR contains the label `area:operator`
- [x] Link this PR to related issues
- [x] I have not made _any_ changes in the `charts/` directory.

### CODE REVIEW

File: .github/workflows/chart-release.yml [hunk 1]
```diff
@@ -30,6 +30,9 @@ jobs:
           echo "CHART_VERSION=$(yq e '.version' charts/provider-postgresql/Chart.yaml)" >> $GITHUB_ENV
           echo "PREVIOUS_CHART_VERSION=$(git show HEAD^:charts/provider-postgresql/Chart.yaml -m | grep "version:" | cut -d ' ' -f 2)" >> $GITHUB_ENV
 
+      - name: Prepare charts
```
Comment (ZuGao) [inline .github/workflows/chart-release.yml:33]: Is there a reason we are changing the chart release pipeline in this PR?

File: apis/postgresql/v1alpha1/postgresql_standalone_types.go [hunk 2]
```diff
@@ -39,6 +39,8 @@ type PostgresqlStandaloneParameters struct {
 
 // PostgresqlStandaloneSpec defines the desired state of a PostgresqlStandalone.
 type PostgresqlStandaloneSpec struct {
+	ConnectableInstance `json:",inline"`
```
Comment (ZuGao) [inline apis/postgresql/v1alpha1/postgresql_standalone_types.go:42]: Can we have more explicit name for this object? for instance `ConnectionInstanceDetails?`

File: apis/postgresql/v1alpha1/connectionsecret.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package v1alpha1
+
+// ConnectableInstance is the composable type for enabling connection secret.
+type ConnectableInstance struct {
```
Comment (ZuGao) [inline apis/postgresql/v1alpha1/connectionsecret.go:4]: Why so many wraps around this string?

File: .github/workflows/chart-release.yml (hunk 1 above)
Comment (ccremer) [reply .github/workflows/chart-release.yml:33]: When merging this PR to master I want to ensure that the CRD is up-to-date and attached to the next helm release. I'm not quite happy myself the way releasing helm charts conflicts a bit with `make generate`, I may come up with a PR that does things a bit better, but for now I say it's required.

File: apis/postgresql/v1alpha1/connectionsecret.go (hunk 3 above)
Comment (ccremer) [reply apis/postgresql/v1alpha1/connectionsecret.go:4]: I made this type composable by intention, to make this type reusable for other API specs in the future, for example `PostgresqlReplicated`. That way we won't have to duplicate `WriteConnectionSecretToRef`.

File: apis/postgresql/v1alpha1/postgresql_standalone_types.go (hunk 2 above)
Comment (ccremer) [reply apis/postgresql/v1alpha1/postgresql_standalone_types.go:42]: Here I'm just composing with a type that has `WriteConnectionSecretRefTo`. The resulting CRD and API spec will have the property `spec.writeConnectionSecretRefTo` as expected when serializing to json or yaml.

File: gen_sample.go [hunk 4]
```diff
@@ -122,6 +122,7 @@ func newPostgresqlStandaloneSample() *v1alpha1.PostgresqlStandalone {
 		},
 		ObjectMeta: metav1.ObjectMeta{Name: "my-instance", Namespace: "default", Generation: 1},
 		Spec: v1alpha1.PostgresqlStandaloneSpec{
+			//ConnectableInstance: v1alpha1.ConnectableInstance{WriteConnectionSecretToRef: v1alpha1.ConnectionSecretRef{Name: "my-secret"}},
```
Comment (Kidswiss) [inline gen_sample.go:125]: Is this commented out intentionally?

Comment (ccremer) [reply gen_sample.go:125]: ah, yes, I wanted that this field is set by the defaulter webhook when installing the sample, so I'll remove the line.

File: .github/workflows/chart-release.yml (hunk 1 above)
Comment (ccremer) [reply .github/workflows/chart-release.yml]: With #39 ff this should not longer be necessary.

**Review by Kidswiss — APPROVED** [review]

**Review by ZuGao — APPROVED** [review]

**Pull request merged**